src/Suave/Cookie.fs

```diff
@@ -42,211 +42,224 @@ module Cookie =
-    | Choice1Of2 cookie_data ->
-      let encoded_data = enc cookie_data
-      { HttpCookie.mk' cookie_name encoded_data
-          with http_only = true
+  let generateCookies serverKey cookieName relativeExpiry secure plainData =
+    let enc, _ = Bytes.cookieEncoding
+    match Crypto.secretbox serverKey plainData with
+    | Choice1Of2 cookieData ->
+      let encodedData = enc cookieData
+      { HttpCookie.mkSimple cookieName encodedData
+          with httpOnly = true
                secure    = secure }
-      |> sliding_expiry relative_expiry
+      |> slidingExpiry relativeExpiry
     | err -> failwithf "internal error on encryption %A" err
 
-  let read_cookies key cookie_name cookies =
-    let _, dec = Bytes.cookie_encoding
+  let readCookies key cookieName cookies =
+    let _, dec = Bytes.cookieEncoding
     let found =
       cookies
-      |> Map.tryFind cookie_name
-      |> Choice.from_option (NoCookieFound cookie_name)
-      |> Choice.map (fun c -> c, c |> (HttpCookie.value >> dec))
+      |> Map.tryFind cookieName
+      |> Choice.from_option (NoCookieFound cookieName)
+      |> Choice.map (fun c -> c, c.value |> dec)
     match found with
-    | Choice1Of2 (cookie, cipher_data) ->
-      cipher_data
-      |> Crypto.secretbox_open key
+    | Choice1Of2 (cookie, cipherData) ->
+      cipherData
+      |> Crypto.secretboxOpen key
       |> Choice.map_2 DecryptionError
-      |> Choice.map (fun plain_text -> cookie, plain_text)
+      |> Choice.map (fun plainText -> cookie, plainText)
     | Choice2Of2 x -> Choice2Of2 x
 
-  let refresh_cookies relative_expiry http_cookie : WebPart =
-    sliding_expiry relative_expiry http_cookie ||> set_pair
-
-  let update_cookies (csctx : CookiesState) f_plain_text : WebPart =
-    context (fun ({ runtime = { logger = logger }} as ctx) ->
-      let plain_text' =
-        match read_cookies csctx.server_key csctx.cookie_name (ctx.response |> HttpResult.cookies) with
-        | Choice1Of2 (_, plain_text) ->
-          Log.log logger "Suave.Cookie.update_cookies" LogLevel.Debug "update_cookies - existing"
-          f_plain_text (Some plain_text)
+  let refreshCookies relativeExpiry httpCookie : WebPart =
+    slidingExpiry relativeExpiry httpCookie ||> setPair
+
+  let updateCookies (csctx : CookiesState) f_plainText : WebPart =
+    context (fun ctx ->
+      let logger = ctx.runtime.logger
+      let plainText =
+        match readCookies csctx.serverKey csctx.cookieName ctx.response.cookies with
+        | Choice1Of2 (_, plainText) ->
+          Log.log logger "Suave.Cookie.updateCookies" LogLevel.Debug "updateCookies - existing"
+          f_plainText (Some plainText)
         | Choice2Of2 _ ->
-          Log.log logger "Suave.Cookie.update_cookies" LogLevel.Debug "update_cookies - first time"
-          f_plain_text None
+          Log.log logger "Suave.Cookie.updateCookies" LogLevel.Debug "updateCookies - first time"
+          f_plainText None
```

Comment:
I actually don't mind this use of "_" to name different function parameters in internal implementation code. Really it should be "fPlainText".  I think it's the fact that the "f" gets harder to see and it's really quite important.